internal/services/authorization/role_management_policy_resource.go

```diff
@@ -0,0 +1,1097 @@
+	additionalRecipients := rule.Values["notificationRecipients"]
+
+	if level != data.NotificationLevel {
+		level = data.NotificationLevel
+	}
+	if defaultRecipients != data.DefaultRecipients {
+		defaultRecipients = data.DefaultRecipients
+	}
+	if recipientChange {
+		additionalRecipients = data.AdditionalRecipients
+	}
+
+	return map[string]interface{}{
+		"id":                         rule.Values["id"],
+		"ruleType":                   rule.Values["ruleType"],
+		"target":                     rule.Values["target"],
+		"recipientType":              rule.Values["recipientType"],
+		"notificationType":           rule.Values["notificationType"],
+		"notificationLevel":          level,
+		"isDefaultRecipientsEnabled": defaultRecipients,
+		"notificationRecipients":     additionalRecipients,
+	}
+}
+
+func flattenNotificationSettings(rule *map[string]interface{}) *RoleManagementPolicyNotificationSettings {
+	var notificationLevel string
+	var defaultRecipients bool
+	var additionalRecipients []string
+
+	if v, ok := (*rule)["notificationLevel"].(string); ok {
+		notificationLevel = v
+	}
+	if v, ok := (*rule)["isDefaultRecipientsEnabled"].(bool); ok {
+		defaultRecipients = v
+	}
+	if v, ok := (*rule)["notificationRecipients"].([]interface{}); ok {
+		additionalRecipients = make([]string, len(v))
+		for i, r := range v {
+			additionalRecipients[i] = r.(string)
+		}
+	}
+	return &RoleManagementPolicyNotificationSettings{
+		NotificationLevel:    notificationLevel,
+		DefaultRecipients:    defaultRecipients,
+		AdditionalRecipients: additionalRecipients,
+	}
+}
+
+func getScopedPolicyId(ctx context.Context, metadata sdk.ResourceMetaData, id *parse.RoleManagementPolicyId, roleDefinitionId string) (*rolemanagementpolicies.ScopedRoleManagementPolicyId, error) {
+	if id.Name != "" {
+		// Check if the current ID is still valid
+		scopedId := pointer.To(rolemanagementpolicies.NewScopedRoleManagementPolicyID(id.Scope, id.Name))
+
+		resp, err := metadata.Client.Authorization.RoleManagementPoliciesClient.Get(ctx, *scopedId)
+		switch {
+		case err != nil && !response.WasNotFound(resp.HttpResponse):
+			return nil, fmt.Errorf("failed to get existing Role Management Policy %s. %+v", scopedId.ID(), err)
+		case resp.Model != nil:
+			return scopedId, nil
+		}
+	}
+
+	// If it wasn't found (or no resource ID was provided), we need to search for the policy ID
+	resp, err := metadata.Client.Authorization.RoleManagementPolicyAssignmentsClient.ListForScope(ctx, commonids.NewScopeID(id.Scope))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list Role Management Policy Assignments for scope %s. %+v", id.Scope, err)
+	}
+
+	for _, assignment := range *resp.Model {
+		if *assignment.Properties.RoleDefinitionId == roleDefinitionId {
+			parts := strings.Split(*assignment.Properties.PolicyId, "/providers/Microsoft.Authorization/roleManagementPolicies/")
+			scopedId := pointer.To(rolemanagementpolicies.NewScopedRoleManagementPolicyID(parts[0], parts[1]))
```

Comment:
Can we use a parser here rather than manually splitting the string? If we have do this, we'll also need to length check `parts` to avoid a potential crash.